wgengine/router/router_windows.go

```diff
@@ -171,75 +211,141 @@ func (ft *firewallTweaker) doAsyncSet() {
-				ft.logf("removed old Tailscale-Process rule in %v", d)
-			}
-			var exe string
-			exe, err = os.Executable()
+		var exe string
+		exe, err = os.Executable()
+		if err != nil {
+			ft.logf("failed to find Executable for Tailscale-Process rule: %v", err)
+		} else {
+			ft.logf("adding Tailscale-Process rule to allow UDP for %q ...", exe)
+			d, err = ft.runFirewall("add", "rule", "name=Tailscale-Process",
+				"dir=in",
+				"action=allow",
+				"edge=yes",
+				"program="+exe,
+				"protocol=udp",
+				"profile=any",
+				"enable=yes",
+			)
 			if err != nil {
-				ft.logf("failed to find Executable for Tailscale-Process rule: %v", err)
+				ft.logf("error adding Tailscale-Process rule: %v", err)
 			} else {
-				ft.logf("adding Tailscale-Process rule to allow UDP for %q ...", exe)
-				d, err = ft.runFirewall("add", "rule", "name=Tailscale-Process",
-					"dir=in",
-					"action=allow",
-					"edge=yes",
-					"program="+exe,
-					"protocol=udp",
-					"profile=any",
-					"enable=yes",
-				)
-				if err != nil {
-					ft.logf("error adding Tailscale-Process rule: %v", err)
-				} else {
-					ft.mu.Lock()
-					ft.didProcRule = true
-					ft.mu.Unlock()
-					ft.logf("added Tailscale-Process rule in %v", d)
-				}
+				ft.mu.Lock()
+				ft.didProcRule = true
+				ft.mu.Unlock()
+				ft.logf("added Tailscale-Process rule in %v", d)
 			}
 		}
-		var err error
-		for _, cidr := range val {
-			ft.logf("adding Tailscale-In rule to allow %v ...", cidr)
-			var d time.Duration
-			d, err = ft.runFirewall("add", "rule", "name=Tailscale-In", "dir=in", "action=allow", "localip="+cidr, "profile=private", "enable=yes")
-			if err != nil {
-				ft.logf("error adding Tailscale-In rule to allow %v: %v", cidr, err)
-				break
-			}
-			ft.logf("added Tailscale-In rule to allow %v in %v", cidr, d)
+	}
+	for _, cidr := range local {
+		ft.logf("adding Tailscale-In rule to allow %v ...", cidr)
+		var d time.Duration
+		d, err := ft.runFirewall("add", "rule", "name=Tailscale-In", "dir=in", "action=allow", "localip="+cidr, "profile=private", "enable=yes")
+		if err != nil {
+			ft.logf("error adding Tailscale-In rule to allow %v: %v", cidr, err)
+			return err
 		}
-		bo.BackOff(ctx, err)
+		ft.logf("added Tailscale-In rule to allow %v in %v", cidr, d)
+	}
 
-		ft.mu.Lock()
-		ft.lastVal = val
-		ft.known = (err == nil)
+	if killswitch && ft.fwProc == nil {
+		exe, err := os.Executable()
+		if err != nil {
+			return err
+		}
+		proc := exec.Command(exe, "/firewall", ft.tunGUID.String())
+		var (
+			out io.ReadCloser
+			in  io.WriteCloser
+		)
+		out, err = proc.StdoutPipe()
+		if err != nil {
+			return err
+		}
+		proc.Stderr = proc.Stdout
+		in, err = proc.StdinPipe()
+		if err != nil {
+			return err
+		}
+
+		go func(out io.ReadCloser) {
+			b := bufio.NewReaderSize(out, 1<<10)
+			for {
+				line, err := b.ReadString('\n')
+				if err != nil {
+					return
+				}
+				line = strings.TrimSpace(line)
+				if line != "" {
+					ft.logf("%s", line)
```

Comment:
using just "%s" makes this rate-limited along with all other "%s" in the program (which Josh has been on a mission to eliminate)

How about "fw-child: %s"?